cypress/integration/test_table.js

```diff
@@ -95,4 +95,86 @@ describe("tests table", function() {
       "_blank"
     );
   });
+
+  describe("Test Status Selector", () => {
+    beforeEach(() => {
+      cy.visit(TESTS_ROUTE);
+      cy.get("#cy-test-status-select > .cy-treeselect-bar").click();
+    });
+
+    it("Status select says 'No filters selected' by default", () => {
+      cy.get("#cy-test-status-select").contains("No filters selected");
+    });
+
+    it("Clicking on 'All' checkbox adds all statuses to URL ", () => {
+      cy.get(".cy-checkbox")
+        .contains("All")
+        .click();
+      cy.location().should(loc => {
+        expect(loc.pathname).to.equal(TESTS_ROUTE);
+        expect(loc.search).to.include("statuses=all,pass,fail,skip,silentfail");
+      });
+    });
+
+    const statuses = [
+      { display: "Pass", key: "pass" },
+      { display: "Silent Fail", key: "silentfail" },
+      { display: "Fail", key: "fail" },
+      { display: "Skip", key: "skip" }
+    ];
+
+    statuses.forEach(({ display, key }) => {
+      it(`Clicking on ${display} status checkbox adds ${key} status to URL and clicking again removes it`, () => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+        cy.location().should(loc => {
+          expect(loc.pathname).to.equal(TESTS_ROUTE);
+          expect(loc.search).to.include(`statuses=${key}`);
+        });
+        cy.wait(100);
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+        cy.location().should(loc => {
+          expect(loc.pathname).to.equal(TESTS_ROUTE);
+          expect(loc.search).to.not.include(`statuses=${key}`);
+        });
+      });
+    });
+
+    it("Checking all statuses will check 'All' checkbox", () => {
+      statuses.forEach(({ display }) => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+      });
+      cy.get(".cy-checkbox > input").should("be.checked");
+    });
+
+    it("Unchecking a status will uncheck 'All' checkbox", () => {
+      statuses.forEach(({ display }) => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+      });
+      cy.get(".cy-checkbox")
+        .contains("Pass")
+        .click();
+      cy.get(".cy-checkbox > input")
+        .first()
+        .should("not.be.checked");
+    });
```

Comment:
these tests are more for the TreeSelect component in general whereas these are the test-table tests

Reply:
I see how these tests are geared towards the TreeSelect because we are testing how TreeSelect actions affect the TreeSelect state. It may make more sense if these tests checked what's happening in the URL instead of the TreeSelect checkbox state after checking/unchecking.

Reply:
what i'm saying is these tests should be for the TreeSelect component and not in the test-table component. otherwise this test would have to be in the patch page table too, which doesn't make much sense.